dags/multipod/configs/mxla_collective_config.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to construct configs for maxtext DAG."""
+
+from xlml.apis import gcp_config, metric_config, task, test_config
+from dags import test_owner, gcs_bucket
+from dags.multipod.configs import common
+from dags.vm_resource import TpuVersion, Project, RuntimeVersion
+import datetime
+
+PROJECT_NAME = Project.CLOUD_ML_AUTO_SOLUTIONS.value
+RUNTIME_IMAGE = RuntimeVersion.TPU_UBUNTU2204_BASE.value
+
+
+def get_mxla_collective_config(
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    bytes_to_transfer: int,
+    project_name: str = PROJECT_NAME,
+    runtime_version: str = RUNTIME_IMAGE,
+    network: str = "default",
+    subnetwork: str = "default",
+    is_tpu_reserved: bool = True,
+    automated_test: bool = True,
+    num_slices: int = 1,
+) -> task.TpuQueuedResourceTask:
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.XLML_DATASET,
+      dataset_project=project_name,
+      composer_project=project_name,
+  )
+
+  current_time = datetime.datetime.now()
+  current_date = current_time.strftime("%Y-%m-%d")
+  current_datetime = current_time.strftime("%Y-%m-%d-%H-%M-%S")
+
+  trigger = "automated" if automated_test else "manual"
+  base_output_directory = f"{gcs_bucket.XLML_OUTPUT_DIR}/multipod/mxla/nightly/{trigger}/{current_date}/{num_slices}slice-V{tpu_version.value}_{tpu_cores}-mxla-collective-{bytes_to_transfer}transferBytes-{current_datetime}"
```

Comment:
cc @raymondzouu in case this is a good fit for the auto-generated bucket names. This PR will land separately, so we can update this once the infra is in place.

Reply:
Yeah the auto-generated bucket name would be useful here if we would like to do some post-processing on the output `benchmark_status.txt` later

Reply:
MXLA team will work on some post-processing on Borg based on the XprofResponse results we get.
I am using the current bucket name as they requested, but I am happy to update in the next PR if needed.

Reply:
Gotcha, thanks Tony. So the benchmark output will be in the form of an XprofResponse. Do you think later on you will want to do some post-processing of the XprofResponse in XLML to upload to our own dashboards or anything like that?